assets/js/Containers/FrevaGPT/index.js

```diff
@@ -319,12 +332,21 @@ class FrevaGPT extends React.Component {
 
           <ChatBlock />
 
+          {this.state.dynamicAnswer ?? (
+            <AnswerComponent
+              content={this.state.dynamicAnswer}
+              variant={this.state.dynamicVariant}
+            />
+          )}
+
           {this.state.loading ? (
```

Comment:
This is a bit cumbersome to read. I think this would be the same logic

```javascript
          {this.state.loading || !this.state.dynamicAnswer ? (
            <Row className="mb-3">
              <Col md={1}>
                <Spinner />
              </Col>
            </Row>
          ) : null}
```

Reply:
This change would lead to a display of the spinner even if there is no request running. The original condition was meant this way: If a request is running (spinner = true) but dynamicAnswer has content (currently streaming) don't show the spinner. So only show spinner when a request is running but the data stream didn't start yet.

Reply:
You are right, how about

```
          {this.state.loading && !this.state.dynamicAnswer ? (
            <Row className="mb-3">
              <Col md={1}>
                <Spinner />
              </Col>
            </Row>
          ) : null}
```